Plugins/Item/Item.cpp

```diff
@@ -217,32 +250,48 @@ ArgumentStack Item::GetEntireItemAppearance(ArgumentStack&& args)
 
 ArgumentStack Item::RestoreItemAppearance(ArgumentStack&& args)
 {
-  ArgumentStack stack;
-
-  if (auto *pItem = item(args))
-    {
-      const auto sAppString = Services::Events::ExtractArgument<std::string>(args);
-      int idx;
-      int stringPos=0;
-
-      for(idx = 0; idx < 6; idx++)
-	{
-          pItem->m_nLayeredTextureColors[idx] = std::stoul(sAppString.substr(stringPos,2), nullptr, 16);
-	  stringPos+=2;
-	}
-
-      for(idx = 0; idx < 3; idx++)
-	{
-          pItem->m_nModelPart[idx] = std::stoul(sAppString.substr(stringPos,2), nullptr, 16);
-	  stringPos+=2;
-	}
-      for(idx = 0; idx < 19; idx++)
-	{
-	  pItem->m_nArmorModelPart[idx] = std::stoul(sAppString.substr(stringPos,2), nullptr, 16);
-	  stringPos+=2;
-	}
-
-    }
-  return stack;
+   ArgumentStack stack;
+
+   if (auto *pItem = item(args))
+     {
+	const auto sAppString = Services::Events::ExtractArgument<std::string>(args);
+	int idx;
+	int stringPos=0;
+	char texture, part;
+	
+	if(sAppString.length()==2*142)
```

Comment:
This is going to break any existing item appearance strings, right? Normally this wouldn't be an issue but GetEntireItemAppearance/RestoreItemAppearance is tailor made for serialisation, and if any modules have used this function already to store item appearances, they will now be lost. Should we add a check for the old string length and replicate the old behaviour here too, to support reading it back in?